cosmo_tester/test_suites/bootstrap_based_tests/distributed_installation_test.py

```diff
@@ -0,0 +1,152 @@
+########
+# Copyright (c) 2018 Cloudify.co Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import time
+import pytest
+
+from cosmo_tester.framework.examples.nodecellar import NodeCellarExample
+from cosmo_tester.framework.test_hosts import \
+    DistributedInstallationCloudifyManager
+from cosmo_tester.framework.util import prepare_and_get_test_tenant
+
+from cosmo_tester.test_suites.snapshots import (
+    create_snapshot,
+    restore_snapshot
+)
+
+USER_NAME = "test_user"
+USER_PASS = "testuser123"
+TENANT_NAME = "tenant"
+
+
+@pytest.fixture(scope='module')
+def database_and_manager(cfy, ssh_key, module_tmpdir, attributes, logger):
+    """
+    Bootstraps a cloudify manager with no DB on a VM in rackspace OpenStack.
+    """
+    # The preconfigure callback populates the files structure prior to the BS
+    def _preconfigure_callback_manager_and_database(database_and_manager):
+        # Updating the database VM first
+        database_and_manager[0].additional_install_config = {
+            'sanity': {'skip_sanity': 'true'},
+            'postgresql_server': {'enable_remote_connections': 'true'},
+            'services_to_install': [
+                'database_service'
+            ]
+        }
+        database_and_manager[1].additional_install_config = {
+            'sanity': {'skip_sanity': 'true'},
+            'postgresql_client': {
+                'host': str(database_and_manager[0].private_ip_address)
+            },
+            'services_to_install': [
+                'queue_service',
+                'composer_service',
+                'manager_service'
+            ]
+        }
+
+    hosts = DistributedInstallationCloudifyManager(cfy=cfy,
+                                                   ssh_key=ssh_key,
+                                                   tmpdir=module_tmpdir,
+                                                   attributes=attributes,
+                                                   logger=logger)
+
+    hosts.instances[1].upload_plugins = False
+
+    hosts.preconfigure_callback = _preconfigure_callback_manager_and_database
+
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+@pytest.fixture(scope='function')
+def nodecellar(cfy, database_and_manager, attributes, ssh_key, tmpdir, logger):
+    # Uploading to the manager not the database
+    manager = database_and_manager[1]
+    manager.use()
```

Comment:
we already do that `.use()` in the `DistributedInstallationCloudifyManager._bootstrap` so it shouldn't be needed here

Anyway, it probably shouldn't be just `[1]`, because we don't want to keep track of this position, when the distributed manager gets more than 2 components. Let's maybe add a property to the `DistributedInstallationCloudifyManager` that returns the component that should be `.use()`'d

Reply:
Great point! Fixed :)